src/worker.cc

```diff
@@ -1142,7 +1144,9 @@ void service_worker_t::set_interrupt(const std::function<void()>* interrupt_func
 }
 void service_worker_t::cleanup() {
   // sends metrics to statsd server over udp
-  statsd_client->flush();
+  if (statsd_client) {
```

Comment:
this is unfortunate. we should have just checked if no host was set here: https://github.com/vthiery/cpp-statsd-client/blob/master/include/cpp-statsd-client/UDPSender.hpp#L184 and returned immedately. then all of this would be unneeded

Reply:
https://github.com/vthiery/cpp-statsd-client/pull/40